src/libsystemd-network/dhcp-option.c

```diff
@@ -93,32 +112,75 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
 
                 options[*offset] = code;
                 options[*offset + 1] = l;
-
                 *offset += 2;
 
                 ORDERED_SET_FOREACH(p, s) {
-                        options[*offset] = p->option;
-                        options[*offset + 1] = p->length;
-                        memcpy(&options[*offset + 2], p->data, p->length);
-                        *offset += 2 + p->length;
+                        r = dhcp_option_append_tlv(options, size, offset, p->option, p->length, p->data);
+                        assert(r >= 0);
+                }
+                break;
+        }
+        case SD_DHCP_OPTION_RELAY_AGENT_INFORMATION: {
+                sd_dhcp_server *server = (sd_dhcp_server *) optval;
+                size_t current_offset = *offset + 2;
+
+                if (server->agent_circuit_id) {
+                        size_t length = strlen(server->agent_circuit_id);
+                        int subcode = SD_DHCP_RELAY_AGENT_CIRCUIT_ID;
```

Comment:
why assign these two to some local variable first?

Reply:
To avoid splitting long line into two

Reply:
there's nothing wrong with adding like breaks. one line per parameter passed is something we often do for functions that take so many args